dask/array/core.py

```diff
@@ -2374,11 +2385,37 @@ def store(x, index, lock):
     slices = slices_from_chunks(arr.chunks)
 
     name = 'store-%s' % arr.name
-    dsk = dict(((name,) + t[1:], (store, t, slc, lock))
+    dsk = dict(((name,) + t[1:], (store, t, slc, lock, region))
                for t, slc in zip(core.flatten(arr._keys()), slices))
     return dsk
 
 
+def storeindask(out, index, x):
+    #Loop through output chunks and setitem:
+    out_sliced = out[index]
+    for slc in slices_from_chunks(out_sliced.chunks):
+        dsk = optimize(out_sliced[slc].dask, out_sliced[slc]._keys())
+        if len(dsk) == 1:
+            for key in dsk:
+                if base_condition(dsk, key):
+                    operator.setitem(dsk[key][1], dsk[key][2], x[slc])
+        elif len(dsk) == 2:
+            for key in dsk:
+                if base_condition(dsk, key):
+                    operator.setitem(dsk[dsk[key][1]], dsk[key][2], x[slc])
+        else:
+            raise NotImplementedError('Only concantenated dask arrays can be stored to')
+
+
+def base_condition(dsk, key):
+    #Simple condition that allows simple setitem:
+    return (isinstance(key, tuple) and
+            isinstance(dsk[key],tuple) and
+            len(dsk[key]) == 3 and
+            dsk[key][0] is getarray and
+            isinstance(dsk[key][2], tuple))
```

Comment:
Generally I'm somewhat afraid of invoking only a part of the dask.array abstraction.  People will say "oh hey, cool, this supports dask.arrays" they'll do lots of crazy things that don't work and then be sad.  The list solution is, I think, a contract that we can more easily fulfill.  It's more clear what does and does not work.

If we want to go down the dask.array route that's fine, but we need to support *all* of it.  I don't know a way to do this reasonably.